Assets/MixedRealityToolkit-Tests/Core/TestFixture_01_MixedRealityToolkitTests.cs

```diff
@@ -0,0 +1,543 @@
+
+            // Add test data provider 1
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider1), new TestExtensionService1("Test04-08-1", 10));
+
+            // Add test data provider 2
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestExtensionService2("Test04-08-2.1", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestExtensionService2("Test04-08-2.2", 10));
+
+            // Retrieve all ITestDataProvider2 services
+            var test2DataProviderServices = MixedRealityToolkit.Instance.GetActiveServices(typeof(ITestDataProvider2));
+
+            // Tests
+            Assert.AreEqual(2, test2DataProviderServices.Count);
+        }
+
+        [Test]
+        public void Test_04_09_GetAllMixedRealityDataProviders()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Add test 1 services
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider1), new TestExtensionService1("Test16-1.1", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider1), new TestExtensionService1("Test16-1.2", 10));
+
+            // Add test 2 services
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestExtensionService2("Test16-2.1", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestExtensionService2("Test16-2.2", 10));
+
+            // Retrieve all extension services.
+            var allExtensionServices = MixedRealityToolkit.Instance.GetActiveServices(typeof(IMixedRealityExtensionService));
+
+            // Tests
+            Assert.AreEqual(4, allExtensionServices.Count);
+        }
+
+        #endregion IMixedRealityDataprovider Tests
+
+        #region IMixedRealityExtensionService Tests
+
+        [Test]
+        public void Test_05_01_RegisterMixedRealityExtensionService()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Register ITestExtensionService1
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestExtensionService1), new TestExtensionService1("Test ExtensionService 1", 10));
+
+            // Retrieve ITestExtensionService1
+            var extensionService1 = MixedRealityToolkit.Instance.GetService(typeof(ITestExtensionService1));
+
+            // Tests
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.AreEqual(1, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+
+            // Tests
+            Assert.IsNotNull(extensionService1);
+        }
+
+        [Test]
+        public void Test_05_02_01_UnregisterMixedRealityExtensionServiceByType()
+        {
+            // Redo register test service for a valid instance.
+            Test_05_01_RegisterMixedRealityExtensionService();
```

Comment:
Rather than re-running a test (which may fail) please consider a helper function that sets the desired initial conditions?